### ISSUE TITLE
Muelu:  Inaccessible path on MueLu_CoalesceDropFactory_def.hpp

### ISSUE THREAD

**Issue by mperrinel** [trilinos/Trilinos#9808]
## Question
@trilinos/@muelu:MueLu_CoalesceDropFactory_def.hpp>

I'm working on the code coverage of MueLu package and I think that one path is not accessible : 
l526:
`  if (algo == "classical") { // l387
         if(classicalAlgo == defaultAlgo) { ...} // l485
         else { // l526 
              /* Cut Algorithm */
              //CMS
              using DropTol = Details::DropTol<real_type,LO>;
              std::vector<DropTol> drop_vec;`  

1) The else path implies that classicalAlgo != defaultAlgo and algo == "classical"
2) The only way to have an algo == "classical" is to enter into the block of code that starts l213 : 
```
 else if(algo == "signed classical" || algo == "block diagonal colored signed classical" || algo == "block diagonal signed classical") {
      useSignedClassical = true;
...
      algo = "classical"; // l231
```
Entering on this block also means that useSignedClassical is set to true.
To resume, when we are at l526, trying to enter in the else condition dedicated to the "Cut Algorithm" we have **classicalAlgo != defaultAlgo** and **useSignedClassical = true**, but in this case, that also means we have got an exception before (l382): 
`      TEUCHOS_TEST_FOR_EXCEPTION(useSignedClassical && classicalAlgo != defaultAlgo, Exceptions::RuntimeError, "\"aggregation: classical algo\" != default is not supported for scalled classical aggregation");`

So from what I understand, it is not possible to enter into the else condition l526, because we have triggered the exception before (l382). It is correct ?

**Comment by cgcgcg** [comment]
Can't we have `algo=="classical"` from line 192, pass through line 272?

**Comment by mperrinel** [comment]
Indeed, you are right, that way, and with the combination of an **classical algo** different from the default one, we can reach this condition. Thanks ! Issue can be closed for me.
